src/main/common/axis.h

```diff
@@ -42,6 +42,4 @@ typedef enum {
     AI_PITCH
 } angle_index_t;
 
-#define ANGLE_INDEX_COUNT 2
-
```

Comment:
it was my suggestion. Angle index is misleading, usually 3 angles are used (roll, pitch, yaw). Using macro hides this fact. 

personally I think that using macros for such trivial and fixed things is making the code harder to read.

At least use something that contains the number, like ANGLE 2D or ANGLE_2D_RP

Reply:
The macro name just follows the naming convention of the enum. I guess the name comes from "angle mode", where yaw is not relevant. Maybe instead of deleting the macro we could rename `angle_index_t` (and the macro) according to your suggestion? For example `anglemode_index_t` and `ANGLEMODE_INDEX_COUNT`

Reply:
Problem is that ANGLEMODE_INDEX_COUNT is obfuscating the fact that it is 2-element array. And it does carry very little extra information (vs `2`) - only that variable is angle-mode related. But that should be already obvious from variable name or context.

It is not a value that is intended to be changed. It will always evaluate to 2 and code will make no sense otherwise.

When 2 (or 2d) is included in its name, it will be only extra typing and I don't have major problem with it. But without it, is making code harder to read and PR review are more difficult (You need to lookup macro meaning to understand code).

(I personally aggree with https://www.kernel.org/doc/html/v4.10/process/coding-style.html. But being a more verbose is fine as long as it is not making code harder to read)

Reply:
I don't mind at all what we do.

I added `#define TWO_AXES 2` to axis.h and that saved 6 bytes of flash space.  Then I tried adding both `#define RP_AXIS_COUNT 2` and on the next line `#defineEF_AXIS_COUNT 2`.

This didn't add any more flash space (maybe compiler points both defines to the same memory address?).

Can easily revert to '2' but it looks OK like this, eg in autopilot.c we have:

```
typedef enum {
    EW = 0,
    NS
} axisEF_t;
```
and later
```
    float pidSumCraft[EF_AXIS_COUNT];
    pt3Filter_t upsample[EF_AXIS_COUNT];
    earthFrame_t efAxis[EF_AXIS_COUNT];
} posHoldState;
```